src/ui/components/Library/Library.tsx

```diff
@@ -41,46 +41,58 @@ function isUnknownWorkspaceId(
 }
 
 function LibraryLoader(props: PropsFromRedux) {
+  const { setModal } = props;
   const auth = useAuth0();
   const { userSettings, loading: userSettingsLoading } = hooks.useGetUserSettings();
   const userInfo = hooks.useGetUserInfo();
   const { workspaces, loading: loading1 } = hooks.useGetNonPendingWorkspaces();
   const { pendingWorkspaces, loading: loading2 } = hooks.useGetPendingWorkspaces();
   const { nags, loading: loading3 } = useGetUserInfo();
+  const dismissNag = hooks.useDismissNag();
 
   useEffect(() => {
     if (!userInfo.loading && !userSettingsLoading) {
       LogRocket.createSession({ userInfo, auth0User: auth.user, userSettings });
     }
   }, [auth, userInfo, userSettings, userSettingsLoading]);
 
+  // TODO [jaril] Fix react-hooks/exhaustive-deps
+  useEffect(function handleOnboardingModals() {
+    if (singleInvitation(pendingWorkspaces?.length || 0, workspaces.length)) {
+      trackEvent("onboarding.team_invite");
+      setModal("single-invite");
+    } else if (downloadReplay(nags, dismissNag)) {
+      trackEvent("onboarding.download_replay_prompt");
+      setModal("download-replay");
+    } else if (firstReplay(nags)) {
+      trackEvent("onboarding.demo_replay_prompt");
+      setModal("first-replay");
+    }
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
looks cleaner, but how does this fix it?

Reply:
Our onboarding modal logic is executed when dashboard is mounted. I just lifted that logic by one level because we have all the data required available to us in `<LibraryLoader />` component. So as soon as we get the user info, we run the onboarding modal logic in the `<LibraryLoader />` component, previously it was getting executed in `<Library />` component. Does this answer your question? Please correct me if I am wrong anywhere 🤔